index.js

```diff
@@ -60,40 +79,73 @@ class Cache {
   }
 }
 
-let _currentSecond
-
-function currentSecond () {
-  if (_currentSecond !== undefined) {
-    return _currentSecond
-  }
-  _currentSecond = Math.floor(Date.now() / 1000)
-  setTimeout(_clearSecond, 1000).unref()
-  return _currentSecond
-}
-
-function _clearSecond () {
-  _currentSecond = undefined
-}
-
 class Wrapper {
-  constructor (func, key, serialize, cacheSize, ttl, onHit) {
-    this.ids = new LRUCache(cacheSize)
-    this.error = null
-    this.started = false
+  // TODO signature
+  constructor (func, key, serialize, references, /*size, */storage, ttl, onDedupe, onHit, onMiss) {
+    // TODO do we want to limit dedupe size?
+    this.dedupes = new Map()
```

Comment:
imo there's no need of LRUCache

Reply:
Switch back to the LRUCache. It's needed otherwise there is a risk of a memory leak.

Reply:
Ah yes. I get it, there is no need for an LRUMap here.